pkg/eventmanager/eventmanager.go

```diff
@@ -20,6 +20,8 @@ const (
 	UPGRADE_EXTDEPCHECK_FAILED_DESC = "Cluster upgrade to version %s was cancelled during the External Dependency Availability Check step. A required external dependency of the upgrade was unavailable, so the upgrade did not proceed. Automated upgrades will be retried on their next scheduling cycle. If you have manually scheduled an upgrade instead, it must now be rescheduled."
 	// UPGRADE_SCALE_FAILED_DESC describes the upgrade scaling failed
 	UPGRADE_SCALE_FAILED_DESC = "Cluster upgrade to version %s was cancelled during the Scale-Up Worker Node step. A temporary additional worker node was unable to be created to temporarily house workloads, so the upgrade did not proceed. Automated upgrades will be retried on their next scheduling cycle. If you have manually scheduled an upgrade instead, it must now be rescheduled."
+	// UPGRADE_UPGRADEABLE_FAILED_DESC describes the upgradeable check failure
+	UPGRADE_UPGRADEABLE_FAILED_DESC = "Cluster upgrade to version %s was cancelled because manual acknowledgement is required concerning the removal of deprecated APIs. Please review the Knowledge Base article at https://access.redhat.com/articles/6329921 and perform the acknowledgement procedure if you are not using workloads that require the deprecated APIs and wish to proceed with the upgrade. Automated upgrades will be retried on their next scheduling cycle. If you have manually scheduled an upgrade instead, it must be rescheduled."
```

Comment:
@Tessg22 @rafael-azevedo the admin ack gate isn't specifically designed to handle API just depreciation. In future releases there will be other related checks. Is there general documentation on these gates in public OCP docs? This doc is specific to 4.9

Reply:
I don't believe there is general documentation as far as I am aware so far.

Reply:
This is a fair point. I don't believe, or at least haven't seen, general documentation around the gates that isn't just the enhancements proposal.

But to solve this specific problem, I think the way to go might be to extract the `Reason` from the `clusterversion` Upgradeable condition and present that as the message here.

For example, for the specific 4.9 issue, the message is:
```
Kubernetes 1.22 and therefore OpenShift 4.9 remove several APIs which require admin consideration. Please see
        the knowledge article https://access.redhat.com/articles/6329921 for details and instructions.
```

So I would be in favour of the eventmanager dynamically generating the failure message based on some wrapper text + the CVO message. 

```
Cluster upgrade to version %s was cancelled because manual acknowledgement of upcoming release changes is required. <<REASON HERE>>. Please perform the acknowledgement procedure if you wish to proceed with an upgrade, then re-schedule your upgrade.  Automated upgrades will be retried on their next scheduling cycle. If you have manually scheduled an upgrade instead, it must be rescheduled.
```

The approach of the `createDelayedDescription()` function, which dynamically generates a delayed message based on the conditions causing the delay, could be modelled upon here. Similar deal, just getting stuff out of CVO instead (use the `cv` package to help)

If you want to give this a go, let me know, otherwise we can add it to the backlog to do before 4.10.

Reply:
@mrbarge did you want to say `createFailureDescription()` instead of `createDelayedDescription()`? We could make use of `failedCondition` var that is already defined at [line 157](https://github.com/openshift/managed-upgrade-operator/blob/edb5b0f9e13109478b22dac3163e8fe3c5edaf16/pkg/eventmanager/eventmanager.go#L157). Then just replace the hardcoded message with `description = failedCondition.Message`. I think there is no need to create a separate card for this. I will test if there are any hidden complications or edge cases.

Reply:
I am wondering if there will be more than one check/condition in the newer versions.

Reply:
@Tessg22 In the future it is a possibility there could be more then 1 ack condition.

Reply:
It's also broader than just acks.  For example, you could be `Upgradeable=False` in the ClusterVersion because you have excessive node skew: openshift/cluster-kube-apiserver-operator#1199.  I'd recommend just passing along any `Upgradeable=False` `reason` and `message`, with wrapping wording drawn from [the API docs][1] if you want to provide more context on the condition (although I'd hope the `message` would speak for itself, and think it's bug-worthy if it does not).

[1]: https://github.com/openshift/api/blob/cc0db1116639638254e87a564902833f1ee006d5/config/v1/types_cluster_operator.go#L179-L189

Reply:
+1 @wking

Reply:
@wking, do you know how multiple "Upgradeable=False" conditions will be displayed in CVO (if at all)?

Reply:
> I'd recommend just passing along any Upgradeable=False reason and message with wrapping wording

What do you think about this wording?
`fmt.Errorf("Cluster upgrade to version %s is canceled with the reason of %s containing message that %s Automated upgrades will be retried on their next scheduling cycle. If you have manually scheduled an upgrade instead, it must be rescheduled.", desiredVersion, condition.Reason, condition.Message)
`

Example: Cluster upgrade to version 1.2.3 is canceled with the reason of IsClusterUpgradable not done containing message that Kubernetes 1.22 and therefore OpenShift 4.9 remove several APIs which require admin consideration. Please see the knowledge article https://access.redhat.com/articles/6329921 for details and instructions. Automated upgrades will be retried on their next scheduling cycle. If you have manually scheduled an upgrade instead, it must be rescheduled.

Reply:
@Tessg22 @anjoshi24 I had a look at the CVO code and yes CVO will [list multiple reasons](https://github.com/openshift/cluster-version-operator/blob/06ec265e3a3bf47b599e56aec038022edbe8b5bb/pkg/cvo/upgradeable.go#L78) if they exist.

Given that, you should be able to just take the `Message` from the CVO condition and use that as the reliable messaging of why the cluster was not upgradeable.

Reply:
In latest commit, I have updated the wording of error message and input of IsClusterUpgradable notification.

Reply:
@Tessg22 That looks good, thank you.